src/ir/control.rs

```diff
@@ -97,6 +97,8 @@ pub struct Invoke {
     pub inst: InstIdx,
     // The ports defined by this invocation
     pub ports: Vec<PortIdx>,
+    /// The events the invocation uses
+    pub events: Vec<EventBind>,
```

Comment:
I think @UnsignedByte also added this so it will cause a conflict with one of the PRs when merged. Hopefully won't be too annoying to fix.

Reply:
Oh Edmund might've done it, but I know last week I also talked to you about this so it might've been me. I sorta implemented this in the mono branch, but when I realized it was wrong (for reasons besides it being an `EventBind` instead of an `EventIdx`) so I implemented it correctly here, so I will just use this version when I merge it